### PR TITLE
Add DirectAssetURL, FilePath & LinkType to Release Links API

### PR DESCRIPTION
**Pull request by 56KBs**
Add optional DirectAssetURL and LinkType for Release Links response data.

Add optional FilePath and LinkType to creation and updating of release links.

See https://docs.gitlab.com/13.10/ee/api/releases/links.html#create-a-link for documentation

### CODE REVIEW

File: releaselinks.go [hunk 1]
```diff
@@ -33,10 +33,12 @@ type ReleaseLinksService struct {
...
-	ID       int    `json:"id"`
-	Name     string `json:"name"`
-	URL      string `json:"url"`
-	External bool   `json:"external"`
+	ID             int    `json:"id"`
+	Name           string `json:"name"`
+	URL            string `json:"url"`
+	External       bool   `json:"external"`
+	DirectAssetURL string `json:"direct_asset_url,omitempty"`
+	LinkType       string `json:"link_type,omitempty"`
```
Comment (svanharmelen) [inline releaselinks.go:41]:
```suggestion
	DirectAssetURL string `json:"direct_asset_url"`
	External       bool   `json:"external"`
	LinkType       string `json:"link_type"`
```

File: fixtures_test.go [hunk 2]
```diff
@@ -68,7 +68,9 @@ const (
 		"id":1,
 		"name":"awesome-v0.2.dmg",
 		"url":"http://192.168.10.15:3000",
-		"external":true
+		"external":true,
+		"direct_asset_url": "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
+		"link_type": "other"
```
Comment (svanharmelen) [inline fixtures_test.go:73]:
```suggestion
		"direct_asset_url": "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
		"external":true,
		"link_type": "other"
```

File: fixtures_test.go [hunk 3]
```diff
@@ -83,7 +85,9 @@ const (
 			"id": 1,
 			"name": "awesome-v0.2.dmg",
 			"url": "http://192.168.10.15:3000",
-			"external": true
+			"direct_asset_url": "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
+			"link_type": "other",
+			"external": false
```
Comment (svanharmelen) [inline fixtures_test.go:90]:
```suggestion
			"direct_asset_url": "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
			"external": false,
			"link_type": "other"
```

File: releaselinks.go [hunk 4]
```diff
@@ -99,8 +101,10 @@ func (s *ReleaseLinksService) GetReleaseLink(pid interface{}, tagName string, li
 //
 // GitLab API docs: https://docs.gitlab.com/ee/api/releases/links.html#create-a-link
 type CreateReleaseLinkOptions struct {
-	Name *string `url:"name" json:"name"`
-	URL  *string `url:"url" json:"url"`
+	Name     *string `url:"name" json:"name"`
+	URL      *string `url:"url" json:"url"`
```
Comment (svanharmelen) [inline releaselinks.go:105]: Not your change, but let's fix it while were add it 😉 
```suggestion
	Name     *string `url:"name,omitempty" json:"name,omitempty"`
	URL      *string `url:"url,omitempty" json:"url,omitempty"`
```

File: releaselinks_test.go [hunk 5]
```diff
@@ -105,15 +162,21 @@ func TestReleaseLinksService_UpdateReleaseLink(t *testing.T) {
...
-		t.Errorf("release link name, expected '%s', got '%s'", exampleReleaseName,
-			releaseLink.Name)
+
+	require.NoError(t, err)
+	expectedRelease := &ReleaseLink{
+		ID:             1,
+		Name:           "awesome-v0.2.dmg",
+		URL:            "http://192.168.10.15:3000",
+		External:       true,
+		DirectAssetURL: "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
```
Comment (svanharmelen) [inline releaselinks_test.go:176]:
```suggestion
		DirectAssetURL: "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
		External:       true,
```

File: releaselinks_test.go [hunk 6]
```diff
@@ -35,40 +38,94 @@ func TestReleaseLinksService_ListReleaseLinks(t *testing.T) {
...
+				"external": false,
+				"link_type": "other"
+			}`,
+			want: &ReleaseLink{
+				ID:             1,
+				Name:           "release-notes.md",
+				URL:            "http://192.168.10.15:3000",
+				DirectAssetURL: "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/docs/release-notes.md",
+				LinkType:       "other",
+				External:       false,
```
Comment (svanharmelen) [inline releaselinks_test.go:109]:
```suggestion
				External:       false,
				LinkType:       "other",
```

File: releaselinks_test.go [hunk 7]
```diff
@@ -35,40 +38,94 @@ func TestReleaseLinksService_ListReleaseLinks(t *testing.T) {
...
+			Name:     "awesome-v0.2.msi",
+			URL:      "http://192.168.10.15:3000/msi",
+			External: true,
+		},
+		{
+			ID:             1,
+			Name:           "awesome-v0.2.dmg",
+			URL:            "http://192.168.10.15:3000",
+			External:       false,
+			DirectAssetURL: "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
```
Comment (svanharmelen) [inline releaselinks_test.go:55]:
```suggestion
			DirectAssetURL: "http://192.168.10.15:3000/namespace/example/-/releases/v0.1/downloads/awesome-v0.2.dmg",
			External:       false,
```

File: releaselinks.go (hunk 1 above)
Comment (svanharmelen) [inline releaselinks.go:41]: I think this needs a `const`, like this for example: https://github.com/xanzy/go-gitlab/blob/master/types.go#L36

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Thanks @56KBs! Do have some comments, mostly making the ordering consistent with the docs and one functional suggestion...

**Comment by 56KBs** [comment]
> Thanks @56KBs! Do have some comments, mostly making the ordering consistent with the docs and one functional suggestion...

Thanks @svanharmelen - I've fixed up the ordering and the other comments :+1:

**Review by svanharmelen — APPROVED** [review]
Thanks @56KBs! LGTM 👍🏻

**Comment by 56KBs** [comment]
@svanharmelen would it be possible to get a release that contains these changes, I would like to make use of these changes within [goreleaser/goreleaser](https://github.com/goreleaser/goreleaser)

**Comment by svanharmelen** [comment]
Just release v0.50.0 👍🏻

**Pull request merged**